Makefile

```diff
@@ -314,7 +314,7 @@ test: $(test-targets)
 unit-test: ## Runs unit test without e2e
 	# Run unit tests, ignoring e2e tests
 	# run the go tests and gen the file coverage-all used to do the integration with codecov
-	SKIP_GINKGO=1 go test -race -covermode=atomic  -coverprofile=unit-coverage.out `go list ./...`
+	SKIP_GINKGO=1 go test -race -covermode=atomic  -coverprofile=unit-coverage.out -coverpkg=./... `go list ./...`
```

Comment:
Unrelated to this PR, but I'm curious why we use `covermode=atomic` for our coverage? IIUC, we are interested in ensuring every line of code is run atleast once during testing (`covermode=set`) rather than counting how many times it ran (see https://go.dev/blog/cover). Is this a mistake? Should we update this to use `covermode=set` setting instead?

Reply:
`-covermode must be "atomic", not "set", when -race is enabled`

Reply:
@azeemshaikh38 Let me know if you still have concerns. If not, I will merge this PR. Thanks

Reply:
Thanks for clarifying! Looks good to merge.